plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteMetadata.java

```diff
@@ -169,10 +168,20 @@ public ConnectorMergeTableHandle beginMerge(ConnectorSession session, ConnectorT
 
         return new IgniteMergeTableHandle(
                 handle,
-                (IgniteOutputTableHandle) mergeTableHandle.getOutputTableHandle(),
+                (IgniteOutputTableHandle) beginInsert(session, new JdbcTableHandle(schemaTableName, remoteTableName, Optional.empty()), ImmutableList.copyOf(columns), retryMode),
                 primaryKeys,
                 columns,
-                mergeTableHandle.getUpdateCaseColumns());
+                updateColumnHandles);
+    }
+
+    @Override
+    public void finishMerge(
```

Comment:
How can you just ignore fragments here? Looks wrong.

Reply:
Ignite not support transactional merge, merge sink(class JdbcMergeSink) operates target table directly.
When step in to this method, the merge is already done, thus don't need to do anything with the pageSinkIds.